spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/SparkTestBaseWithCatalog.java

```diff
@@ -87,7 +87,7 @@ public SparkTestBaseWithCatalog(
     config.forEach(
         (key, value) -> spark.conf().set("spark.sql.catalog." + catalogName + "." + key, value));
 
-    if (config.get("type").equalsIgnoreCase("hadoop")) {
+    if (null != config.get("type") && config.get("type").equalsIgnoreCase("hadoop")) {
```

Comment:
It is usually better to reverse the equals so that the method call is on the constant:

```java
if ("hadoop".equalsIgnoreCase(config.get("type"))) {...}
```

That way it's always null-safe.